sql/certificates/certs.go

```diff
@@ -52,6 +52,28 @@ func GetHareOutput(db sql.Executor, lid types.LayerID) (types.BlockID, error) {
 	return result, nil
 }
 
+func FirstInEpoch(db sql.Executor, epoch types.EpochID) (types.BlockID, error) {
+	var (
+		result types.BlockID
+		err    error
+		rows   int
+	)
+	if rows, err = db.Exec(`
+		select block from certificates where layer between ?1 and ?2 and valid = 1
```

Comment:
do we keep track of invalid certificates? thats a waste

Reply:
we don't keep track of those that were invalid in the first place.
we only keep certificate that was valid before but later became invalid.
1. hare output but has no certificate to back it up (this db entry will have certificate as null)
2. a certificate that was valid when received but no longer valid when the 2nd valid certificate is received.